From a17d95bd8a95c714bc04e3f4e32acb983df2f8c0 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Thu, 27 May 2021 18:49:40 +0300
Subject: [PATCH 69/89] iot-gate-imx8: dts: add wakeup sources

Add device tree for wakeup sources.
The current wakeup sources are the input GPIOs of the digital I/O add-on
module.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile         |  1 +
 .../dts/compulab/sb-iotgimx8-ied-wakeup.dts   | 59 +++++++++++++++++++
 2 files changed, 60 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-wakeup.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 49b7cda91995..051abbb16b47 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -14,6 +14,7 @@ dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied-m4.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-poed.dtb
 dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-poedv2.dtb
+dtb-$(CONFIG_ARCH_MXC) += sb-iotgimx8-ied-wakeup.dtb
 
 always-y         := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-wakeup.dts b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-wakeup.dts
new file mode 100644
index 000000000000..db1bf5775a4a
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8-ied-wakeup.dts
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2021 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include <dt-bindings/input/gpio-keys.h>
+#include "sb-iotgimx8-ied.dts"
+
+/ {
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		 in0 {
+			label = "in0";
+			gpios = <&gpio3 0 GPIO_ACTIVE_HIGH>;
+			linux,input-type = <EV_KEY>;
+			linux,code = <KEY_0>;
+			wakeup-source;
+			debounce-interval = <1000>;
+			wakeup-event-action = <EV_ACT_ASSERTED>;
+		};
+		in1 {
+			label = "in1";
+			gpios = <&gpio3 1 GPIO_ACTIVE_LOW>;
+			linux,input-type = <EV_KEY>;
+			linux,code = <KEY_1>;
+			wakeup-source;
+			debounce-interval = <1000>;
+			wakeup-event-action = <EV_ACT_DEASSERTED>;
+		};
+		in2 {
+			label = "in2";
+			gpios = <&gpio3 6 GPIO_ACTIVE_LOW>;
+			linux,input-type = <EV_KEY>;
+			linux,code = <KEY_2>;
+			wakeup-source;
+			debounce-interval = <1000>;
+			wakeup-event-action = <EV_ACT_DEASSERTED>;
+		};
+		in3 {
+			label = "in3";
+			gpios = <&gpio3 7 GPIO_ACTIVE_LOW>;
+			linux,input-type = <EV_KEY>;
+			linux,code = <KEY_3>;
+			wakeup-source;
+			debounce-interval = <1000>;
+			wakeup-event-action = <EV_ACT_DEASSERTED>;
+		};
+	};
+};
-- 
2.17.1

